### DISCUSSION TITLE
Make views and fields available for use

### DISCUSSION THREAD

**Discussion by HoneySpace** [payloadcms/payload#1729] [Feature Requests & Ideas]
### Idea
Make components from `payload/components/views` and `payload/components/fields` available for use inside of custom components and pages.

### Use case
Database have two collections A and B
```ts
const A: CollectionConfig = {
  slug: "A",
  fields: [
    {
      name: "title",
      type: "text",
      label: "Title",
    },
    {
      name: "description",
      type: "textarea",
      label: "Description",
    },
  ],
};

const B: CollectionConfig = {
  slug: "A",
  fields: [
    {
      name: "a",
      label: "A",
      type: "relationship",
      relationTo: A.slug,
    },
    {
      name: "name",
      type: "text",
      label: "Name",
    },
    {
      name: "tag",
      type: "text",
      label: "Tag",
    },
  ],
};
```
I need to display a table of records form B collection related to an A record in Edit view . 

It would be nice to be able to create UI field and just import List view or  Cell component from payload, so the table would have the same styling, preferences support, and that would save time, since no need to write own implimentation for rendering fields

### Problem
Right now if I try to import components from `payload/components/views` and `payload/components/fields` webpack will throw following error
```
./node_modules/payload/components/views/Cell.ts
Module parse failed: Unexpected token (2:7)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export { default as Cell } from '../../dist/admin/components/views/collections/List/Cell';
> export type { Props } from '../../dist/admin/components/views/collections/List/Cell/types';
```

**Comment by jacobsfletch** [comment]
@HoneySpace 💯 this is absolutely going to happen. We're actively laying the groundwork for this right now and it's going to be great. While tables are not available yet, some _fields are_. The SEO plugin is a good example of this by extending Payload's text, textarea, and upload fields. Check it out: https://github.com/payloadcms/plugin-seo/tree/main/src/fields

**Reply by HoneySpace** [reply]
@jacobsfletch Glad to hear that! Having some of fields are helping a lot 🚀
Absolutly love the fact that Payload lets you use custom components and extend existing ones

**Reply by wherewego-nina** [reply]
@jacobsfletch  Any idea when this might be available? Specifically the views?

**Comment by jacobsfletch** [comment]
Related discussion: https://github.com/payloadcms/payload/discussions/1249
